packages/alfa-aria/src/feature.ts

```diff
@@ -57,12 +57,20 @@ export class Feature<N extends string = string> {
 }
 
 export namespace Feature {
-  export type Aspect<T> = Mapper<Element, T>;
+  export type Aspect<T, U extends unknown = unknown> = Mapper<
+    Element,
+    T,
+    Array<U>
+  >;
```

Comment:
```suggestion
  export type Aspect<T, A extends Array<unknown> = []> = Mapper<
    Element,
    T,
    A
  >;
```

This way, multiple additional arguments can be defined if needed.

Reply:
I do not understand the difference… 🤔 
Actually, I do not understand either why we have 
```javascript
export type Mapper<T, U = T, A extends Array<unknown> = Array<unknown>> = (
  value: T,
  ...args: A
) => U;
```
instead of
```javascript
export type MapperBis<T, U = T, A extends unknown = unknown> = (
  value: T,
  ...args: Array<A>
) => U;
```

what are the use-cases allowed by one and not the other?

```javascript
function f(x: number, ...args: Array<string>): boolean { return args.length === x }
```
is both a `Mapper<number, Array<string>, boolean>` and a `MapperBis<number, string, boolean>`.

---
I feel there is something in `A extends Array<unknown>` being different from `Array<X extends unknown>`… Not sure what however…

Reply:
It may be easiest to illustrate with an example. Let's compare this...

```ts
type P<T> = (...args: Array<T>) => void;
```

...to this:

```ts
type Q<A extends Array<unknown>> = (...args: A) => void;
```

If I want a function that takes _one_ argument of type `number`, `Q<[number]>` would expand to `(a: number) => void`. If I want a function that takes _two_ arguments of type `number` followed by type `string`, `Q<[number, string]>` would expand to `(a: number, b: string) => void`.

In both of those cases, the best I can do with `P` is `P<number>` and `P<number | string>` and both would expand to variadic functions that take `number` and `number |  string` arguments, respectively. `Q` works by relying on _tuples_ rather than arrays and this is how we can control exactly which arguments the expanded function has.

Reply:
> `Q<[number, string]>` would expand to `(a: number, b: string) => void`.

Ah, that's it!
I thought that rest parameters must have the same type and that `function f(...args: [number, string])` wasn't allowed…
Makes sense now! 💡